src/views/UserRole/table.js

```diff
@@ -0,0 +1,293 @@
+    return a[1] - b[1];
+  });
+  return stabilizedThis.map((el) => el[0]);
+}
+const headCells = [
+  { id: "userId", numeric: true, disablePadding: true, label: "UserId" },
+  { id: "names", numeric: true, disablePadding: false, label: "names" },
+  { id: "Email", numeric: true, disablePadding: false, label: "Email" },
+  { id: "Role", numeric: true, disablePadding: false, label: "Role" },
+  { id: "Action", numeric: true, disablePadding: false, label: "Action" },
+];
+const useStyles = makeStyles((theme) => ({
+  root: {
+    width: "100%",
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(1),
+  },
+  paper: {
+    width: "100%",
+    marginBottom: theme.spacing(55),
+  },
+  table: {
+    minWidth: 750,
+    minHeight: "400px"
+  },
+  visuallyHidden: {
+    border: 0,
+    clip: "rect(0 0 0 0)",
+    height: 1,
+    margin: -1,
+    overflow: "hidden",
+    padding: 0,
+    position: "absolute",
+    top: 20,
+    width: 1,
+  },
+  highlight:
+    theme.palette.type === "light"
+      ? {
+          color: theme.palette.secondary.main,
+          backgroundColor: lighten(theme.palette.secondary.light, 0.85),
+        }
+      : {
+          color: theme.palette.text.primary,
+          backgroundColor: theme.palette.secondary.dark,
+        },
+  title: {
+    flex: "1 1 100%",
+  },
+}));
+export default function EnhancedTable() {
+  const classes = useStyles();
+  const [order, setOrder] = React.useState("asc");
+  const [orderBy, setOrderBy] = React.useState("UserId");
+  const [page, setPage] = React.useState(0);
+  const [rowsPerPage, setRowsPerPage] = React.useState(5);
+  const handleRequestSort = (event, property) => {
+    const isAsc = orderBy === property && order === "asc";
+    setOrder(isAsc ? "desc" : "asc");
+    setOrderBy(property);
+  };
+  const [users, setUsers] = useState([]);
+  useEffect(() => {
+    const getUsers = async () => {
+      try {
+        const response = await fetch("http://localhost:5000/user/");
+        const JSONData = await response.json();
+        const data = JSONData.user;
+        setUsers(data);
+        // console.log(data);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    getUsers();
+  }, []);
+  const[assignRole, setAssignRole] = useState(null);
+  const [roles, setRoles] = useState([]);
+  useEffect(() => {
+    const getRoles = async () => {
+      try {
+        const response = await fetch("http://localhost:5000/roles/");
+        const JSONData = await response.json();
+        const data = JSONData.roles;
+        setRoles(data);
+      } catch (error) {
+        console.log(error);
+      }
+    };
+    getRoles();
+  }, []);
+  const updateRole = async (userId, roleId) => {
+    try {
+        const data = {
+          _userId: userId.toString(),
+          _roleId: roleId.toString(), 
+        };
+        await fetch("http://localhost:5000/roles/assign", {
+        method: 'post',
+        headers: {
```

Comment:
This will not work in production, please consider storing `http://localhost:5000` in a variable in `.env` so that you call that variable jus to make it dynamic  to let this work in either development or production.

Reply:
I have done setting the environment variable!